bgpd/bgpd.c

```diff
@@ -5304,6 +5304,13 @@ int peer_timers_connect_set(struct peer *peer, uint32_t connect)
 	peer->connect = connect;
 	peer->v_connect = connect;
 
+	if (!CHECK_FLAG(peer->sflags, PEER_STATUS_GROUP)
+	    && (peer->status != Established)) {
```

Comment:
Can you move additional checks below under?:
```
if (!CHECK_FLAG(peer->sflags, PEER_STATUS_GROUP))
        return 0;
```
It's the same check actually (partial).